controllers/openstackmachine_controller.go

```diff
@@ -439,6 +439,11 @@ func (r *OpenStackMachineReconciler) getOrCreate(logger logr.Logger, cluster *cl
 	}
 
 	if instanceStatus == nil {
+		if openStackMachine.Spec.InstanceID != nil {
+			logger.Info("Not reconciling machine in failed state. The previously existing OpenStack instance is no longer available")
+			conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.InstanceNotFoundReason, clusterv1.ConditionSeverityError, "virtual machine no longer exists")
+			return nil, nil
```

Comment:
I wonder if we shouldn't return error in that case? It's not a successful reconciliation loop. On the other hand retrying it doesn't really make sense either.

Reply:
I think the behaviour should be analogous to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/6020a9e69aecf97c43505ae2a577a4d3be8bd903/controllers/openstackmachine_controller.go#L292 state.

Reply:
Makes perfect sense, thanks!